Patch-Source: https://sources.debian.org/src/ncurses/6.4-4/debian/patches/debian-env-access.diff/
we use this to patch CVE-2023-29491,
by making disable-root-environ functionally equivalent to disable-setuid-environ
(which only exists in latest ncurses git).

that new option fixes the issue.
--
Author: Sven Joachim <svenjoac@gmx.de>
Description: Change the --disable-root-environ configure option behavior
 By default, the --disable-root-environ option forbids program run by
 the superuser to load custom terminfo entries.  This patch changes
 that to only restrict programs running with elevated privileges,
 matching the behavior of the --disable-setuid-environ option
 introduced in the 20230423 upstream patchlevel.
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1034372#29
Bug: https://lists.gnu.org/archive/html/bug-ncurses/2023-04/msg00018.html
Forwarded: not-needed
Last-Update: 2023-05-01

---
 ncurses/tinfo/access.c |    2 --
 1 file changed, 2 deletions(-)

--- a/ncurses/tinfo/access.c
+++ b/ncurses/tinfo/access.c
@@ -215,8 +215,6 @@ _nc_env_access(void)
 
     if (is_elevated()) {
 	result = FALSE;
-    } else if ((getuid() == ROOT_UID) || (geteuid() == ROOT_UID)) {
-	result = FALSE;
     }
     return result;
 }

